src/cpu/jit_uni_eltwise.cpp

```diff
@@ -883,22 +839,260 @@ struct jit_uni_relu_kernel_f32 : public jit_uni_eltwise_kernel_f32,
+            // load full Vmm size
+            uni_vmovups(vr_from, mem_from);
+            if (is_bwd())
+                uni_vmovups(vr_for_comparison, mem_for_comparison);
+
+        } else {
+            // load exactly one data item
+            movss(Xmm(vr_from.getIdx()), mem_from);
+            if (is_bwd())
+                movss(Xmm(vr_for_comparison.getIdx()), mem_for_comparison);
+        }
+    }
+
+    // Load 8bit data type (s8)
+    void load_8bit(const bool vectorize, const Vmm& vr_from, const Address& mem_from,
+        const Vmm& vr_for_comparison, const Address& mem_for_comparison) {
+
+        // data type s8 load as s32
+        if (vectorize) {
+            // load full Vmm size
+            if (isa == sse41)
+                pmovsxbd(vr_from, mem_from);
+            else
+                vpmovsxbd(vr_from, mem_from);
+            if (is_bwd()) {
+                if (isa == sse41)
+                    pmovsxbd(vr_for_comparison, mem_for_comparison);
+                else
+                    vpmovsxbd(vr_for_comparison, mem_for_comparison);
+            }
+        } else {
+            // load exactly one data item
+            mov(reg_s8.cvt8(), mem_from);
+            movsx(reg_s8.cvt32(), reg_s8.cvt8());
+            movq(Xmm(vr_from.getIdx()), reg_s8);
+            if (is_bwd()) {
+                mov(reg_s8.cvt8(), mem_for_comparison);
+                movsx(reg_s8.cvt32(), reg_s8.cvt8());
+                movq(Xmm(vr_for_comparison.getIdx()), reg_s8);
+            }
+        }
+    };
+
+    // Load vregs with data from mem
+    void load(const bool vectorize, const Vmm& vr_from, const Address& mem_from,
+        const Vmm& vr_for_comparison, const Address& mem_for_comparison) {
+
+        // Branching on data size
+        if (is32bit())
+            load_32bit(vectorize, vr_from, mem_from, vr_for_comparison, mem_for_comparison);
+        else
+            load_8bit(vectorize, vr_from, mem_from, vr_for_comparison, mem_for_comparison);
+    }
+
+    // Processing
+    void process(const Vmm& vr_to, const Vmm& vr_from, const Vmm& vr_for_comparison);
+
+    // Store 32bit (f32, s32) for any isa
+    void store_32bit(const bool vectorize, const Address& mem_to, const Vmm& vr_to) {
+        if (vectorize) {
+            // store full Vmm size
+            uni_vmovups(mem_to, vr_to);
+        } else {
+            // store exactly one data item
+            movss(mem_to, Xmm(vr_to.getIdx()));
+        }
+    }
+
+    // Store 8bit (s8)
+    void store_8bit(const bool vectorize, const Address& mem_to, const Vmm& vr_to);
+
+    // Store results from vregs to mem
+    void store(const bool vectorize, const Address& mem_to, const Vmm& vr_to) {
+        // Branching on data size
+        if (is32bit())
+            store_32bit(vectorize, mem_to, vr_to);
+        else
+            store_8bit(vectorize, mem_to, vr_to);
+    }
+
+    void compute_step(bool vectorize, const size_t uf, const size_t shift) {
```

Comment:
Since all three steps `load` ,`process` and `store` need to be specialized for `int`.  I think it will be much simpler to have an independent structure for int relu kernel something like `struct jit_uni_relu_kernel_int : public jit_uni_eltwise_kernel, public jit_generator {..};`